### DISCUSSION TITLE
Set custom HTML property in XAML for WASM app

### DISCUSSION THREAD

**Discussion by andrewkdci** [unoplatform/uno#13949] [Q&A] [answered]
I would like to set a custom HTML property in XAML for my WASM app. For example, I would like to set the [data-*](https://www.w3schools.com/Tags/att_global_data.asp) tag.

I looked into [Platform-specific XAML markup](https://platform.uno/docs/articles/platform-specific-xaml.html). 

I added the following namespace to my Page: `xmlns:wasm="http://uno.ui/wasm"`

and added it to the ignorable list: `mc:Ignorable="d wasm" `

and then tried to modify a text block with the custom property: `<TextBlock wasm:data-mydata="this is my custom data"/>`

but it doesn't work.

Any ideas how one might do this?

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the question. XAML does not support setting native attributes this way. You'll need to use the [`SetHtmlAttribute` method](https://github.com/unoplatform/uno/blob/4e4c25ca08e8a1c0eb417bfe38a06edf6ff50ab7/src/Uno.UI.Runtime.WebAssembly/Xaml/UIElementWasmExtensions.cs#L87C22-L87C38) in order to do so. The problem at this time is that since `net8.0-browser` is not to be included in the .NET 8, this makes it difficult to get cross-platform use of those particular APIs in the current shared library structure.

The only way at this time to have access to those APIs is do use dependency injection to provide an interface (defined in the app's class library) implementation (defined in the Wasm head) that will take a `UIElement` and allow you to invoke `SetHtmlAttribute` on it to set the data tag you need.

**Reply by andrewkdci** [reply]
[SetHtmlAttribute](https://github.com/unoplatform/uno/blob/4e4c25ca08e8a1c0eb417bfe38a06edf6ff50ab7/src/Uno.UI.Runtime.WebAssembly/Xaml/UIElementWasmExtensions.cs#L87C22-L87C38) was exactly what I was looking for. Thanks!
